### PR TITLE
agent: refactor guest hooks

### PR DESCRIPTION
**Pull request by YushuoEdge**
We have to execute some hooks both in host and guest. And in /libs/kata-sys-util/src/hooks.rs, the coomon operations are implemented.

In this commit, we are going to refactor the code of guest hooks using code in /libs/kata-sys-util/src/hooks.rs.

Fixes:#5857

Signed-off-by: Yushuo <y-shuo@linux.alibaba.com>

### CODE REVIEW

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -1099,12 +1101,13 @@ impl BaseContainer for LinuxContainer {
...
         }
 
-        if spec.hooks.is_some() {
-            info!(self.logger, "poststop");
-            let hooks = spec.hooks.as_ref().unwrap();
-            for h in hooks.poststop.iter() {
-                execute_hook(&self.logger, h, &st).await?;
-            }
+        // guest Poststop hook
+        // Hook path: guest agent namespace
```
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1105]: Can you rewrite these two lines of comments easy to understand?

File: src/agent/rustjail/src/container.rs [hunk 2]
```diff
@@ -1378,15 +1378,13 @@ async fn join_namespaces(
         info!(logger, "notify child parent ready to run prestart hook!");
         read_async(pipe_r).await?;
 
-        info!(logger, "get ready to run prestart hook!");
```
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1381]: Why delete this line?

File: src/tools/runk/libcontainer/src/container.rs [hunk 3]
```diff
@@ -144,9 +146,9 @@ impl Container {
                 .hooks
                 .as_ref()
                 .ok_or_else(|| anyhow!("hooks config was not present"))?;
-            for h in hooks.poststop.iter() {
-                container::execute_hook(logger, h, &oci_state).await?;
-            }
+            info!(&logger, "Poststop Hooks");
```
Comment (liubin) [inline src/tools/runk/libcontainer/src/container.rs:149]: It's better to refactor these lines using `if let Some()` like https://github.com/kata-containers/kata-containers/pull/5858/files#diff-5bf1516db5499202e6844836d7ebe31e26e2ded8ceec5536618409873a7fa3a1R1159-R1162

File: src/agent/rustjail/src/container.rs [hunk 4]
```diff
@@ -1566,143 +1564,6 @@ fn set_sysctls(sysctls: &HashMap<String, String>) -> Result<()> {
...
-    });
-
-    match tokio::time::timeout(Duration::new(timeout, 0), join_handle).await {
-        Ok(r) => r.unwrap(),
-        Err(_) => Err(anyhow!(nix::Error::ETIMEDOUT)),
-    }
-}
-
-// valid environment variables according to https://doc.rust-lang.org/std/env/fn.set_var.html#panics
-fn valid_env(e: &str) -> Option<(&str, &str)> {
```
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1683]: `valid_env` is required, and should be moved to `libs`.

File: src/agent/rustjail/src/container.rs (hunk 1 above)
Comment (YushuoEdge) [reply src/agent/rustjail/src/container.rs]: Sorry, fixed.

File: src/agent/rustjail/src/container.rs (hunk 2 above)
Comment (YushuoEdge) [reply src/agent/rustjail/src/container.rs:1381]: Sorry, I thought the two logs before and after have expressed the same meaning. Fixed.

File: src/tools/runk/libcontainer/src/container.rs (hunk 3 above)
Comment (YushuoEdge) [reply src/tools/runk/libcontainer/src/container.rs:145]: fixed.

File: src/agent/rustjail/src/container.rs (hunk 4 above)
Comment (YushuoEdge) [reply src/agent/rustjail/src/container.rs:1683]: I am using [OsString](https://doc.rust-lang.org/std/ffi/struct.OsString.html) to represent same meaning. Like in: https://github.com/kata-containers/kata-containers/blob/main/src/libs/kata-sys-util/src/hooks.rs#L207-L213

Comment (liubin) [reply src/agent/rustjail/src/container.rs:1683]: Can OsString check strings containing NUL(`\0`)?

**Review by bergwolf — APPROVED** [review]
lgtm!

**Comment by liubin** [comment]
@YushuoEdge you have a not well-formated commit, you can squash it into the first one.

**Review by studychao — APPROVED** [review]
LGTM

**Comment by YushuoEdge** [comment]
@liubin I have updated this PR, PTAL!

**Pull request merged**